recipe-server/normandy/recipes/tests/test_api.py

```diff
@@ -488,6 +488,46 @@ def test_list_filter_text(self, api_client):
         for recipe in res.data:
             assert recipe['id'] in [r1.id, r2.id]
 
+    def test_update_recipe_action(self, api_client):
+        r = RecipeFactory()
+        a = ActionFactory(name='test')
+
+        res = api_client.patch(f'/api/v1/recipe/{r.pk}/', {'action': 'test'})
+        assert res.status_code == 200
+
+        r.refresh_from_db()
+        assert r.action == a
+
+    def test_update_recipe_locale(self, api_client):
+        r = RecipeFactory()
+        l = LocaleFactory(code='en-US')
+
+        res = api_client.patch(f'/api/v1/recipe/{r.pk}/', {'locales': ['en-US']})
+        assert res.status_code == 200
+
+        r.refresh_from_db()
+        assert r.locales.filter(pk=l.pk).exists()
```

Comment:
Could we assert against a list of locales instead of just checking if the one exists? As it is this will pass if other countries remain in the locale list, which isn't what we want, I think.

Might even want to set up another locale as already assigned to the recipe, to be sure that it's removing the old locales as well as adding the new ones.